spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/core/PubSubTemplate.java

```diff
@@ -119,6 +150,70 @@ public Subscriber subscribe(String subscription, MessageReceiver messageHandler)
 		return subscriber;
 	}
 
+	/**
+	 * Pulls messages on demand using the pull request in argument.
+	 *
+	 * <p>
+	 * After messages are received, this method sends acknowledgements for the received messages if
+	 * any was received.
+	 */
+	@Override
+	public List<PubsubMessage> pull(PullRequest pullRequest) {
```

Comment:
As I understand, there are two ways to do a pull: (1) with a simple timeout and (2) with complicated retry settings. I would argue that (1) is the most common/useful one, and we should make it easy for the developer to specify the timeout without going through the convoluted builders.

One possible way of doing it is to get rid of the whole pull settings option, and accept a timeout in pull methods directly: `pull(PullRequest pullRequest, long timeoutMs)`.

Another alternative is to introduce a configuration property like `spring.cloud.gcp.pubsub.pullTimeoutMs` and inject it into the `SubscriptionAdminSettings` in the auto-config.

WDYT?

Reply:
There indeed might be a default `pullTimeout` option for the method like `pull(PullRequest pullRequest)`, but I'd really consider auto-config on the matter in the separate issue/pull request.

Looks like we don't have consensus for the original API yet, so no reason to overcomplicate.

Reply:
+1 for the `spring.cloud.gcp.pubsub.pullTimeoutMillis` property.

@artembilan wdyt?

Reply:
As I said: I don't want to hijack original API PR for auto-configuration story. 
We have to come back to it in the separate issue/PR.